### DISCUSSION TITLE
[Type] cannot be serialized

### DISCUSSION THREAD

**Discussion by speller26** [pydantic/pydantic#5319] [Question] [answered]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

I want to serialize a model that uses the `Type` type for some fields, but such models cannot be serialized. The included code snippet is directly copied from the [documentation](https://docs.pydantic.dev/usage/types/#type), with the only modification being the addition of a `.json()` call.

### Example Code

```Python
from typing import Type

from pydantic import BaseModel
from pydantic import ValidationError


class Foo:
    pass


class Bar(Foo):
    pass


class Other:
    pass


class SimpleModel(BaseModel):
    just_subclasses: Type[Foo]


SimpleModel(just_subclasses=Foo).json()

"""
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [1], in <cell line: 23>()
     19 class SimpleModel(BaseModel):
     20     just_subclasses: Type[Foo]
---> 23 SimpleModel(just_subclasses=Foo).json()

File ~/opt/anaconda3/lib/python3.9/site-packages/pydantic/main.py:490, in pydantic.main.BaseModel.json()

File ~/opt/anaconda3/lib/python3.9/json/__init__.py:234, in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    232 if cls is None:
    233     cls = JSONEncoder
--> 234 return cls(
    235     skipkeys=skipkeys, ensure_ascii=ensure_ascii,
    236     check_circular=check_circular, allow_nan=allow_nan, indent=indent,
    237     separators=separators, default=default, sort_keys=sort_keys,
    238     **kw).encode(obj)

File ~/opt/anaconda3/lib/python3.9/json/encoder.py:199, in JSONEncoder.encode(self, o)
    195         return encode_basestring(o)
    196 # This doesn't pass the iterator directly to ''.join() because the
    197 # exceptions aren't as detailed.  The list call should be roughly
    198 # equivalent to the PySequence_Fast that ''.join() would do.
--> 199 chunks = self.iterencode(o, _one_shot=True)
    200 if not isinstance(chunks, (list, tuple)):
    201     chunks = list(chunks)

File ~/opt/anaconda3/lib/python3.9/json/encoder.py:257, in JSONEncoder.iterencode(self, o, _one_shot)
    252 else:
    253     _iterencode = _make_iterencode(
    254         markers, self.default, _encoder, self.indent, floatstr,
    255         self.key_separator, self.item_separator, self.sort_keys,
    256         self.skipkeys, _one_shot)
--> 257 return _iterencode(o, 0)

File ~/opt/anaconda3/lib/python3.9/site-packages/pydantic/json.py:97, in pydantic.json.pydantic_encoder()

TypeError: Object of type 'type' is not JSON serializable
"""
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.9.0
            pydantic compiled: True
                 install path: /Users/speller26/opt/anaconda3/lib/python3.9/site-packages/pydantic
               python version: 3.9.12 (main, Apr  5 2022, 01:52:34)  [Clang 12.0.0 ]
                     platform: macOS-13.2.1-arm64-arm-64bit
     optional deps. installed: ['dotenv', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [X] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment, marked as answer]
We don't by default have a JSON encoder for types - as there's no obvious easy to encode types.

As per the docs, you can set up your own custom encoder to do this.
